app/src/main/java/com/example/sharewhatyoucanproject/DonorActivity.kt

```diff
@@ -0,0 +1,233 @@
+        // Checking whether FilePathUri Is empty or not.
+        if (FilePathUri != null) {
+
+            // Setting progressDialog Title.
+            progressDialog!!.setTitle("Image is Uploading...")
+
+            // Showing progressDialog.
+            progressDialog!!.show()
+
+            // Creating second StorageReference.
+            val storageReference2nd = storageReference!!.child(
+                Storage_Path + System.currentTimeMillis() + "." + GetFileExtension(FilePathUri)
+            )
+
+            // Adding addOnSuccessListener to second StorageReference.
+            storageReference2nd.putFile(FilePathUri!!)
+                .addOnSuccessListener { taskSnapshot -> // Getting image name and description from EditText and store into string variable.
+                    val TempImageName = ImageName!!.text.toString().trim { it <= ' ' }
+                    val TempImageDesc = ImageDescription!!.text.toString().trim { it <= ' ' }
+                    // float TempLocation = LocationButton.getVa
+
+                    // Hiding the progressDialog after done uploading.
+                    progressDialog!!.dismiss()
+
+                    // Showing toast message after done uploading.
+                    Toast.makeText(
+                        applicationContext,
+                        "Image Uploaded Successfully ",
+                        Toast.LENGTH_LONG
+                    ).show()
+                    val imageUploadInfo = ImageUploadInfo(
+                        TempImageName,
+                        TempImageDesc,
+                        taskSnapshot.uploadSessionUri.toString()
+                    )
+
+                    // Getting image upload ID.
+                    val ImageUploadId = databaseReference!!.push().key
+
+                    // Adding image upload id s child element into databaseReference.
+                    databaseReference!!.child(ImageUploadId!!).setValue(imageUploadInfo)
+                } // If something goes wrong .
+                .addOnFailureListener { exception -> // Hiding the progressDialog.
+                    progressDialog!!.dismiss()
+
+                    // Showing exception error message.
+                    Toast.makeText(this@DonorActivity, exception.message, Toast.LENGTH_LONG).show()
+                } // On progress change upload time.
+                .addOnProgressListener(object : OnProgressListener<UploadTask.TaskSnapshot?> {
+                    override fun onProgress(snapshot: UploadTask.TaskSnapshot) {
+
+                        // Setting progressDialog Title.
+                        progressDialog!!.setTitle("Image is Uploading...")
+                    }
+                })
+        } else {
+            Toast.makeText(
+                this@DonorActivity,
+                "Please Select Image or Add Image Name",
+                Toast.LENGTH_LONG
+            ).show()
+        }
```

Comment:
@nithishrajapuram3 and @gangamamulya, let's chat about how to reformat this, so that it's more production quality.

Reply:
Sure professor, I have tried from my side to reformat the code as much as I can. Since I'm new to kotlin, I tried it in the best possible way to reformat. Could you please suggest me if anything else that can be done? Going forward I will try to produce more efficient code.